### PR TITLE
feat: add geolocation modal

### PR DESCRIPTION
**Pull request by pyphilia**
Screenshot 2024-03-25 at 17 08 21

### CODE REVIEW

File: src/components/item/settings/GeolocationModalButton.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  const { t } = useBuilderTranslation();
+  const { t: commonT } = useCommonTranslation();
+  const [open, setOpen] = useState(false);
+  const { data: geoloc } = hooks.useItemGeolocation(item.id);
+  const { mutate: saveGeoloc } = mutations.usePutItemGeolocation();
+
+  const helperLabelRef = useRef<HTMLInputElement>(null);
+  const addressLabelRef = useRef<HTMLInputElement>(null);
+  const latRef = useRef<HTMLInputElement>(null);
+  const lngRef = useRef<HTMLInputElement>(null);
```
Comment (spaenleh) [inline src/components/item/settings/GeolocationModalButton.tsx:37]: Any reason why we use refs and not controlled inputs ?

File: src/langs/fr.json [hunk 2]
```diff
@@ -340,5 +340,15 @@
   "ITEM_SETTINGS_IS_COLLAPSED_TITLE": "Minifier un élément",
   "ITEM_SETTINGS_DESCRIPTION_PLACEMENT_ABOVE": "Au-dessus du contenu",
   "ITEM_SETTINGS_DESCRIPTION_PLACEMENT_BELOW": "En-dessous du contenu",
-  "ITEM_SETTINGS_DESCRIPTION_PLACEMENT_TITLE": "Placement de la description"
+  "ITEM_SETTINGS_DESCRIPTION_PLACEMENT_TITLE": "Placement de la description",
+  "ITEM_GEOLOCATION_ADVANCED_BUTTON": "Localisation avancée",
+  "ITEM_GEOLOCATION_ADVANCED_MODAL_TITLE": "Paramètres avancés pour la localisation",
+  "ITEM_GEOLOCATION_ADVANCED_MODAL_DESCRIPTION": "Aucune information soumise par ce formulaire ne sera validée. L'exactitude des données revient à l'éditeur.",
```
Comment (spaenleh) [inline src/langs/fr.json:346]: "Les informations fournies via ce formulaire ne sont pas validées. Vérifiez l'exactitude des données."

**Review by spaenleh — APPROVED** [review]
Thank you for this PR.
Would it make sens to put the country dropdown before the address ? so we can go from less specific to most specific ? 

Can users input only lat+long or the address ? How can we make it clear that it is an OR choice and that the other parts will get autofilled (if that is the case) ?

**Review by spaenleh — APPROVED** [review]
Thank you for this PR.
Would it make sens to put the country dropdown before the address ? so we can go from less specific to most specific ? 

Can users input only lat+long or the address ? How can we make it clear that it is an OR choice and that the other parts will get autofilled (if that is the case) ?

**Review by spaenleh — APPROVED** [review]
Thank you for this PR.
Would it make sens to put the country dropdown before the address ? so we can go from less specific to most specific ? 

Can users input only lat+long or the address ? How can we make it clear that it is an OR choice and that the other parts will get autofilled (if that is the case) ?

File: src/components/item/settings/GeolocationModalButton.tsx (hunk 1 above)
Comment (pyphilia) [reply src/components/item/settings/GeolocationModalButton.tsx:37]: Isn't it best for performance? If I use controlled buttons I also need `useEffect` on values.

**Pull request merged**